torch/csrc/distributed/rpc/message.h

```diff
@@ -128,7 +128,8 @@ TORCH_API Message createExceptionResponse(const std::exception& e, int64_t id);
 TORCH_API Message
 createExceptionResponse(const std::string& exceptionStr, int64_t id);
 
-typedef torch::utils::Future<Message> FutureMessage;
+using FutureMessage = torch::utils::Future<Message>;
+using FutureIValue = torch::utils::Future<at::IValue>;
```

Comment:
Any reason we are sticking with `torch::utils::Future` instead of using `c10::ivalue::Future`? Not sure what the agreement here is on eventually merging these - the way I understood it is that `torch::utils::Future` should be the generic future used across all of PT, and then the existing `c10::ivalue::Future` is just a `torch::utils::Future` that holds an `IValue`. Is this still the plan (cc @xush6528 @wanchaol )

Reply:
@rohan-varma there's still bunch of things we need to do (i.e. make APIs consistent) before really adopting this plan. For the use of here, I think we could just use `c10::ivalue::Future` here as it's now just holding a IValue instead of a generic Message type.

Reply:
Will switch to `ivalue::Future` in a followup PR. I wanted to make this PR as short as possible, as this has impacts on several different places, e.g., rref/profiler/python-frontend.

Reply:
If we want to pass a ivalue future to a profiler jit operator that takes future, we do need to merge them.

For the Python RPC API, we can return a ivalue::Future instead, which is set by callback of Future<Message>. and kill the PyThonFuture wrapper in rpc/init.cpp. It requires a bit more code change. It's ok to do it in the next PR.